api/v1/egressgateway_types.go

```diff
@@ -0,0 +1,277 @@
+
+	// AWS defines the additional configuration options for Egress Gateways on AWS.
+	// +optional
+	AWS *AWSEgressGateway `json:"aws,omitempty"`
+}
+
+// EgressGatewayDeploymentPodSpec is the Egress Gateway Deployment's PodSpec.
+type EgressGatewayDeploymentPodSpec struct {
+	// Affinity is a group of affinity scheduling rules for the EGW pods.
+	// +optional
+	Affinity *v1.Affinity `json:"affinity,omitempty"`
+
+	// NodeSelector gives more control over the nodes where the Egress Gateway pods will run on.
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// TerminationGracePeriodSeconds defines the termination grace period of the Egress Gateway pods in seconds.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	TerminationGracePeriodSeconds *int64 `json:"terminationGracePeriodSeconds,omitempty"`
+
+	// TopologySpreadConstraints defines how the Egress Gateway pods should be spread across different AZs.
+	// +optional
+	TopologySpreadConstraints []v1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
+}
+
+// EgressGatewayDeploymentPodTemplateSpec is the EGW Deployment's PodTemplateSpec
+type EgressGatewayDeploymentPodTemplateSpec struct {
+	// Metadata is a subset of a Kubernetes object's metadata that is added to
+	// the pod's metadata.
+	// +optional
+	Metadata *EgressGatewayMetadata `json:"metadata,omitempty"`
+
+	// Spec is the EGW Deployment's PodSpec.
+	// +optional
+	Spec *EgressGatewayDeploymentPodSpec `json:"spec,omitempty"`
+}
+
+// EgressGatewayMetadata contains the standard Kubernetes labels and annotations fields.
+type EgressGatewayMetadata struct {
+	// Labels is a map of string keys and values that may match replica set and
+	// service selectors. Each of these key/value pairs are added to the
+	// object's labels provided the key does not already exist in the object's labels.
+	// If not specified will default to projectcalico.org/egw:<name>, where <name> is
+	// the name of the Egress Gateway resource.
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Annotations is a map of arbitrary non-identifying metadata. Each of these
+	// key/value pairs are added to the object's annotations provided the key does not
+	// already exist in the object's annotations.
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+type EgressGatewayIPPool struct {
+	// Name is the name of the IPPool that the Egress Gateways can use.
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// CIDR is the IPPool CIDR that the Egress Gateways can use.
+	// +optional
+	CIDR string `json:"cidr,omitempty"`
+}
+
+// NativeIP defines if Egress Gateway pods should have AWS IPs.
+// When NativeIP is enabled, the IPPools should be backed by AWS subnet.
+type NativeIP string
+
+const (
+	NativeIPEnabled  NativeIP = "Enabled"
+	NativeIPDisabled NativeIP = "Disabled"
+)
+
+type LogLevel string
+
+const (
+	LogLevelTrace LogLevel = "Trace"
```

Comment:
Do we actually have a meaningful distinction between Trace and Debug? Most of our components do not, so I think I'd recommend dropping "Trace" and just using "Debug" to mean the most verbose log level.

Reply:
EGW pod has some logs which are trace and some which are debug. Can I just leave ``Trace``` as it is.

Reply:
Gotcha